systems/plants/test/urdf_manipulator_dynamics_test.cpp

```diff
@@ -0,0 +1,55 @@
+
+
+#include <iostream>
+#include <Eigen/Dense>
+#include "URDFRigidBodyManipulator.h"
+
+using namespace std;
+
+int main(int argc, char* argv[])
+{
+	if (argc<2) {
+		cerr << "Usage: urdf_kin_test urdf_filename" << endl;
+		exit(-1);
+	}
+  URDFRigidBodyManipulator* model = loadURDFfromFile(argv[1]);
+
+//  URDFRigidBodyManipulator* model = loadURDFfromFile("/Users/russt/drc/software/models/mit_gazebo_models/mit_robot/model.urdf");
+  if (!model) {
+  	cerr << "ERROR: Failed to load model from " << argv[1] << endl;
+  	return -1;
+  }
+
+  // the order of the bodies may be different in matlab, so print it out once here
+  cout << model->num_bodies << endl;
+  for (int i=0; i<model->num_bodies; i++) {
+    cout << model->bodies[i]->linkname << endl;
+  }
+
+  // run kinematics with second derivatives 100 times
+  VectorXd q = VectorXd::Zero(model->num_dof);
+	VectorXd v = VectorXd::Zero(model->num_velocities);
+	int i;
+
+  if (argc>=2+model->num_dof) {
+  	for (i=0; i<model->num_dof; i++)
+  		sscanf(argv[2+i],"%lf",&q(i));
+  }
+
+	if (argc>=2+model->num_dof+model->num_velocities) {
+		for (i=0; i<model->num_velocities; i++)
+			sscanf(argv[2+model->num_dof+i],"%lf",&v(i));
+  }
+
+  model->doKinematicsNew(q.data(), true, v.data(), true);
+
+  auto H = model->massMatrix<double>();
+	cout << H.value() << endl;
+
+  map<int, unique_ptr<GradientVar<double, TWIST_SIZE, 1>> > f_ext;
+  auto C = model->inverseDynamics(f_ext);
+  cout << C.value() << endl << endl;
+
+  delete model;
+  return 0;
+}
```

Comment:
This file needs a formatting fix. If you've imported the formatter file for Eclipse (https://github.com/tkoolen/drake/wiki/Eclipse-setup-(experimental)#importing-formatter-settings), it's just cmd+shift+F. If you just want to fix the indentation, select all and press cmd+I.